Fix Bugzilla 363.  Enable ATA subsystem clock before writing to ATA
subsystem.

cvs diff -puN -r1.2 -r1.3 arch/arm/mach-mx3/mx31ads_gpio.c
Index: arch/arm/mach-mx3/mx31ads_gpio.c
===================================================================
RCS file: /cvs/local/ltib/imx27ads/linux-2.6.19/arch/arm/mach-mx3/mx31ads_gpio.c,v
retrieving revision 1.2
retrieving revision 1.3
diff -p -u -r1.2 -r1.3
--- linux-2.6.19.2/arch/arm/mach-mx3/mx31ads_gpio.c	15 Jan 2007 05:50:46 -0000	1.2
+++ linux-2.6.19.2-mx/arch/arm/mach-mx3/mx31ads_gpio.c	22 Jan 2007 21:20:34 -0000	1.3
@@ -1051,6 +1051,9 @@ void gpio_ata_active(void)
 	 */
 	mxc_iomux_set_pad(MX31_PIN_USBH2_STP, PAD_CTL_PKE_NONE);	// ATA_DMARQ
 	mxc_iomux_set_pad(MX31_PIN_USBH2_CLK, PAD_CTL_PKE_NONE);	// ATA_INTRQ
+
+	printk(KERN_DEBUG "gpio_ata_active: Enable clocks\n");
+	mxc_clks_enable(ATA_CLK);
 }
 
 EXPORT_SYMBOL(gpio_ata_active);
@@ -1061,6 +1064,9 @@ EXPORT_SYMBOL(gpio_ata_active);
  */
 void gpio_ata_inactive(void)
 {
+	printk(KERN_DEBUG "gpio_ata_inactive: Disable clocks\n");
+	mxc_clks_disable(ATA_CLK);
+
 	/*
 	 * Turn off ATA group B signals
 	 */
cvs diff -puN -r1.1 -r1.2 drivers/ide/arm/mxc_ide.c
Index: drivers/ide/arm/mxc_ide.c
===================================================================
RCS file: /cvs/local/ltib/imx27ads/linux-2.6.19/drivers/ide/arm/mxc_ide.c,v
retrieving revision 1.1
retrieving revision 1.2
diff -p -u -r1.1 -r1.2
--- linux-2.6.19.2/drivers/ide/arm/mxc_ide.c	15 Jan 2007 05:44:49 -0000	1.1
+++ linux-2.6.19.2-mx/drivers/ide/arm/mxc_ide.c	22 Jan 2007 21:20:34 -0000	1.2
@@ -1010,21 +1010,19 @@ static int __init mxc_ide_init(void)
 	printk(KERN_INFO
 	       "MXC: IDE driver, (c) 2004-2006 Freescale Semiconductor\n");
 
-	ATA_RAW_WRITE(MXC_IDE_CTRL_ATA_RST_B, MXC_IDE_ATA_CONTROL);
-
 	/* Select group B pins, and enable the interface */
-
 	PBC_ATA_SIGNAL_ACTIVE();
-	/* Configure the pads */
 
+	/* Configure the pads */
 	gpio_ata_active();
 
-	/* Set initial timing and mode */
+	/* Deassert the reset bit to enable the interface */
+	ATA_RAW_WRITE(MXC_IDE_CTRL_ATA_RST_B, MXC_IDE_ATA_CONTROL);
 
+	/* Set initial timing and mode */
 	set_ata_bus_timing(4, PIO);
 
 	/* Reset the interface */
-
 	mxc_ide_resetproc(NULL);
 
 	/*
